Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1958,3 +1960,102 @@ acceptScheduledBooking (personId, merchantId, _) clientId bookingId = do
   mbActiveSearchTry <- QST.findActiveTryByQuoteId booking.quoteId
   void $ acceptStaticOfferDriverRequest mbActiveSearchTry driver booking.quoteId Nothing merchantId clientId -- handle driver blocked
   pure Success
+
+clearDriverFeeWithCreate ::
+  (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) =>
+  (Id SP.Person, Id DM.Merchant, Id DMOC.MerchantOperatingCity) ->
+  ServiceNames ->
+  (HighPrecMoney, HighPrecMoney, HighPrecMoney) ->
+  DDF.FeeType ->
+  Currency ->
+  Maybe SPayment.DeepLinkData ->
+  m ClearDuesRes
+clearDriverFeeWithCreate (personId, merchantId, opCityId) serviceName (fee, cgst, sgst) feeType currency mbDeepLinkData = do
+  dueDriverFee <- QDFE.findAllByStatusAndDriverIdWithServiceNameFeetype personId [DDF.PAYMENT_PENDING] feeType serviceName
+  driverFee <-
+    case dueDriverFee of
+      [] -> do
+        driverFee' <- mkDriverFee
+        QDF.create driverFee'
+        pure [driverFee']
+      dfee -> pure dfee
+  invoices <- mapM (\fee_ -> runInReplica (QINV.findActiveManualInvoiceByFeeId fee_.id Domain.PAYOUT_REGISTRATION_INVOICE Domain.ACTIVE_INVOICE)) driverFee
```

Comment:
`invoices <- mapM (\fee_ -> runInReplica (QINV.findActiveManualInvoiceByFeeId fee_.id Domain.PAYOUT_REGISTRATION_INVOICE Domain.ACTIVE_INVOICE)) driverFee`
should be 
`invoices <- mapM (\fee_ -> runInReplica (QINV.findActiveManualInvoiceByFeeId fee_.id (feeTypeToInvoicetype feeType) Domain.ACTIVE_INVOICE)) driverFee`